### DISCUSSION TITLE
Request feature

### DISCUSSION THREAD

**Discussion by GrimDawn-max** [Lailloken/Exile-UI#87] [Ideas & Feedback]
GGG have a performance display feature accessed by pressing F1. It displays a whole bunch of data on performance but this isn't stored anywhere. Would be great if you could build an AHK script that collects that data in a file in real time, for later analysis, as either a csv or text file, until you press a hot key or exit the game.

**Comment by Lailloken** [comment]
Which part of the data would you want to store: FPS, ping, hardware utilization, or all of it? It's certainly possible to read it from screen-data, though it kind of takes unnecessary detours by doing so. These graphs are also not positioned in the same place all the time (opening the inventory moves them to the left, for example), so there would be blank space in the collected data. The only thing that's feasible in terms of required effort and resulting gains is reading the client.txt for the ip-address of the instance-server and pinging it to log connection-quality. There are tools that are far better suited to log the other type of data (and maybe even ping too).

**Comment by GrimDawn-max** [comment]
Ideally I’d like to collect the cpu, gpu, latency, server, memory, vram, driver and shader values, which are all below the graph. Don’t really care about fps. The info in client.txt isn’t of any value (imho). I know what you mean about the graph positions moving. Very annoying. 

If you know of other programs that would be better suited to capturing the info off the screen let me know what they are and I’ll take a look ! Still think it’s be interesting if you could get it working, at least to some degree in AHK !

**Reply by Lailloken** [reply]
> capturing the info off the screen

That's where the problem lies. It's super inefficient to try to read these stats from the screen, not to mention the performance cost. You would gain the ability to log your performance-stats at the cost of (possibly) crippling your performance. Depending on the hardware, screen reading can take a lot of CPU cycles and therefore cause stutter and/or low FPS. It's just not worth it.

Additionally, the bars are semi-transparent (even at the highest setting), which makes it pretty much impossible to screen-read reliably and would be even heavier on performance (because the bar-color would constantly shift slightly, and reading color ranges basically multiplies CPU load).

In the Reddit chat, I already suggested using HWinfo64 to log everything hardware-related (CPU, GPU, RAM, VRAM). It won't be able to log shader load or ping but it's a start.
